apps/prairielearn/assets/scripts/instructorAssessmentManualGradingAssessmentQuestionClient.ts

```diff
@@ -468,8 +464,68 @@ onDocumentReady(() => {
       }
     },
   });
+
+  // Build an internal state of filter selection to avoid reading from the html again on click
+  const rubricFilterState: Record<string, boolean> = Object.fromEntries(
+    (rubric_data?.rubric_items ?? []).map((item) => [item.id, false]),
+  );
+
+  document.querySelectorAll('.js-rubric-item-filter').forEach((checkbox) =>
+    checkbox.addEventListener('change', (e) => {
+      rubricFilterState[(e.target as HTMLInputElement).value] =
+        !rubricFilterState[(e.target as HTMLInputElement).value];
```

Comment:
Nit: while this is probably correct, we want to treat the DOM as the source of truth since this isn't a React-style controlled component. You probably want something like this instead:


```suggestion
      if (!(e.target instanceof HTMLInputElement)) return;
      rubricFilterState[e.target.value] = e.target.checked;
```